priv/test_files/unused_callbacks/warnings.erl

```diff
@@ -0,0 +1,25 @@
+-module(warnings).
+
+-callback unused_callback(map()) -> boolean().
+-callback used_callback(map()) -> boolean().
+
+-export([unused_callback/1, call_callback/1]).
+
+%% This function doesn't count as callback usage because
+%% it's just a regular module function that happens to have
+%% the same name as the callback definition, but it may have
+%% nothing to do with the callback itself like in this example
+%% in where `other_module:call/2` is "opaque" to us and its
+%% implementation may not use this module callbacks at all
+unused_callback(#{module := Module, state := State}) ->
+    other_module:call(Module, State).
+
+%% This function counts as callback usage (although it really isn't)
+%% because it's being called from `call_callback/1` and Hank detects
+%% the atom `used_callback` in it, although `other_module:do/1`
+%% implementation may not use this module callbacks at all
```

Comment:
This is not _exactly_ the truth. What counts as _usage_ is not the function itself. What counts as usage is the invocation, within `call_callback/1`. I would remove this comment and, if you really have to, add a similar but more precise comment to `call_callback/1`.
But… This is a justification of an edge case which is a _false negative_ (i.e. a callback that is unused but Hank can't detect). Personally, I would not put this in a test (i.e. I would not validate that Hank _can't_ find something). Or… if you really really want to validate that… I would put the comment on the callback itself (line 4 here) with something like…

```erlang
%% @todo [#…] Detect that this callback is actually unused.
-callback used_callback(…
```

Reply:
But to be clear: I'm 100% sure that we'll never be able to get rid of that `@todo` while maintaining dialyzer-levels of certainty™️.

Reply:
Removed the comment to avoid confusion!